sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -592,6 +597,47 @@ public GlobalThroughputControlConfigBuilder createGlobalThroughputControlConfigB
         return new GlobalThroughputControlConfigBuilder(this, databaseId, containerId);
     }
 
+    void openConnectionsAndInitCaches() {
+        blockListVoidResponse(openConnectionsAndInitCachesInternal());
+    }
+
+    private Mono<List<Void>> openConnectionsAndInitCachesInternal() {
+        int concurrency = 1;
+        int prefetch = 10;
+        if (this.proactiveContainerInitConfig != null) {
+            return Flux.fromIterable(this.proactiveContainerInitConfig.getCosmosContainerIdentities())
+                    .flatMap(
```

Comment:
we are opening the containers sequentially, I think this should be the way as we would never want to overload the metadata request limit

Reply:
Are you saying reduce prefetch as well - makes sense. Done. flatmap by default - without specifying isn't processing sequentially on FlatMap - it uses Concurrency of 256 and prefetch of 32 by default.